app/src/main/java/com/epfl/drawyourpath/userProfile/UserModel.kt

```diff
@@ -0,0 +1,362 @@
+     */
+    fun setActivityTimeGoal(activityTimeGoal: Double): CompletableFuture<Boolean>{
+        checkActivityTimeGoal(activityTimeGoal)
+        return database.setActivityTimeGoal(userId, activityTimeGoal).thenApply {
+            if(it){
+                this.activityTimeGoal=activityTimeGoal
+            }
+            it
+        }
+    }
+
+    /**
+     * Get the daily number of paths goal of the user
+     * @return daily number of paths goal of the user
+     */
+    fun getNumberOfPathsGoal(): Int{
+        return nbOfPathsGoal
+    }
+
+    /**
+     * Use this function to modify the daily number of paths goal of the user
+     * @param nbOfPaths new daily number of paths goal
+     */
+    fun setNumberOfPathsGoal(nbOfPathsGoal: Int): CompletableFuture<Boolean>{
+        checkNbOfPathsGoal(nbOfPathsGoal)
+        return database.setNbOfPathsGoal(userId, nbOfPathsGoal).thenApply {
+            if(it){
+                this.nbOfPathsGoal=nbOfPathsGoal
+            }
+            it
+        }
+    }
+
+    /**
+     * Get the age of the user
+     * @return the age of the user
+     */
+    fun getAge(): Int{
+        return ChronoUnit.YEARS.between(dateOfBirth, LocalDate.now()).toInt()
+    }
+
+    /**
+     * This function will remove the user with username to the friend list
+     * @param username of the user that we want to remove
+     */
+    fun removeFriend(username: String) {
+        if(!friendsList.contains(username)) {
+            throw java.lang.Error("This user is not in the friend list !")
+        }
+        friendsList.remove(username)
+        //TODO: Update the friend list in the database
+    }
+
+    /**
+     * This function will add the user with username to the friend list. To be added the user must be present in the database.
+     * @param username of the user that we want to add to the friend list
+     */
+    fun addFriend(username: String) {
+        //TODO:this function will be implemented during a next task when the database will be cleaned
+        //TODO: Update the friend list in the database
+    }
+
+    /**
+     * This function will return the friend list of a user
+     * @return the friend list of the user
+     */
+    fun getFriendList():Map<String, String> {
+        return this.friendsList
+    }
+
+}
+
+/**
+ * Helper function to check that the firstname and the surname respect the name format condition of the app
+ * @param name to be checked
+ * @return true if the name is in the correct format, and false otherwise
+ */
+private fun checkNameFormat(name: String): Boolean{
+    if(name.find { !it.isLetter() && it != '-' } != null || name.isEmpty()){
+        return false
+    }
+    return true
+}
+
+/**
+ * Helper function to check if the firstname format is correct and throw directly an error if it is incorrect
+ * @param firstanme to be check
+ * @return true if the format is correct and throw an error otherwise
+ */
+private fun checkFirstname(firstname: String): Boolean{
```

Comment:
`checkFirstname` and `checkSurname` are exactly the same methods, they should be merged in a single one.